nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -0,0 +1,680 @@
+                                                                      final String subject, final String input, final String message) {
+        return validationResult
+                .explanation(
+                        String.format("Kinesis Client Library Configuration property with name with%s cannot be used with value \"%s\" : %s",
+                                StringUtils.capitalize(subject), input, message)
+                )
+                .valid(false).build();
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
+        final Set<ValidationResult> validationResults = new HashSet<>(super.customValidate(validationContext));
+
+        initialPositionInStream = InitialPositionInStream.valueOf(StringUtils.trimToEmpty(validationContext.getProperty(INITIAL_STREAM_POSITION).getValue()));
+        try {
+            dateTimeFormatter = DateTimeFormatter.ofPattern(validationContext.getProperty(TIMESTAMP_FORMAT).evaluateAttributeExpressions().getValue());
+        } catch (IllegalArgumentException iae) {
+            validationResults.add(new ValidationResult.Builder().valid(false)
+                    .subject(TIMESTAMP_FORMAT.getName())
+                    .explanation(String.format("%s must be a valid java.time.DateTimeFormatter format", TIMESTAMP_FORMAT.getDisplayName()))
+                    .build()
+            );
+        }
+
+        if (InitialPositionInStream.AT_TIMESTAMP == initialPositionInStream) {
+            if (!validationContext.getProperty(STREAM_POSITION_TIMESTAMP).isSet()) {
+                validationResults.add(new ValidationResult.Builder().valid(false)
+                        .subject(STREAM_POSITION_TIMESTAMP.getName())
+                        .explanation(String.format("%s must be provided when %s is %s", STREAM_POSITION_TIMESTAMP.getDisplayName(),
+                                INITIAL_STREAM_POSITION.getDisplayName(), AT_TIMESTAMP.getDisplayName()))
+                        .build()
+                );
+            } else if (dateTimeFormatter != null) {
+                final String streamTimestamp = validationContext.getProperty(STREAM_POSITION_TIMESTAMP).getValue();
+                try {
+                    dateTimeFormatter.parse(streamTimestamp);
+                    startStreamPositionTimestamp = new Date(
+                            LocalDateTime.parse(validationContext.getProperty(STREAM_POSITION_TIMESTAMP).getValue(), dateTimeFormatter)
+                                    .toInstant(ZoneOffset.UTC).toEpochMilli()
+                    );
+                } catch (Exception e) {
+                    validationResults.add(new ValidationResult.Builder().valid(false)
+                            .subject(STREAM_POSITION_TIMESTAMP.getName())
+                            .explanation(String.format("%s must be parsable by %s", STREAM_POSITION_TIMESTAMP.getDisplayName(),
```

Comment:
Typo: parseable (?)

Reply:
The oh-so-reliable [Wikipedia](https://en.wiktionary.org/wiki/parsable) suggests both versions of the word are correct and "without the e" generally prefered (in formal language at least) - IntelliJ complains at me if I add the `e` so I'll leave as-is unless you really object :slightly_smiling_face:

Reply:
Interesting, my IntelliJ is complaining about "parsable" :)
Let's leave it as it is now.

Reply:
Good ol' i18n - I'm in UK, guess you're US (or another English speaking locale)?

Reply:
Yes, the proofreading language has been set to English (USA) in my IntelliJ.